src/modules/commander/commander.cpp

```diff
@@ -1757,12 +1758,13 @@ int commander_thread_main(int argc, char *argv[])
 	int32_t ef_time_thres = 1000.0f;
 	uint64_t timestamp_engine_healthy = 0; /**< absolute time when engine was healty */
 
-	int32_t disarm_when_landed = 0;
+	float disarm_when_landed = 0;
```

Comment:
Minor: should be `= 0.f`

In https://github.com/PX4/Firmware/blob/91b923a3db445611c3eeeab15551de9a32d31d87/src/modules/commander/commander.cpp#L1862
This will first cast to uint64_t, dropping any decimal places, and then do the multiplication. It should be:
```
(hrt_abstime)(disarm_when_landed * 1000000.f)
```

And in https://github.com/PX4/Firmware/blob/91b923a3db445611c3eeeab15551de9a32d31d87/src/modules/commander/commander.cpp#L2247
It should be made explicit that it's a float:
```
auto_disarm_hysteresis.set_hysteresis_time_from(false, disarm_when_landed * 1000000.f);  
```